### PR TITLE
chore(utils): move useEndlessScrollList to juno-ui-components, depreacate juno-utils package

### PR DESCRIPTION
**Pull request by barsukov**
# Summary

This pull request deprecates the `juno-utils` package, transfers the most essential hook to `@cloudoperators/juno-ui-components`, and updates its usage within our applications (`greenhouse`, `supernova`, `example`, and `doop`).

# Changes Made

- Deprecated the `juno-utils` package (but did not delete it). Cause it is better to create a new one as separated PR which going to be only focus on mock-server lib
- Transferred the critical hook from `juno-utils` to `@cloudoperators/juno-ui-components`.
- Updated all instances of the hook in the following applications:
  - `greenhouse`
  - `supernova`
  - `example`
  - `doop`

# Related Issues

- #446 

# Testing Instructions

1. Run `npm i` to ensure dependencies are up-to-date.
2. Start the application with `turbo dev --filter=supernova`.
3. Navigate to the "Alerts" section.
4. Attempt to scroll down within the "Alerts" section to verify functionality.

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.

### CODE REVIEW

File: packages/ui-components/src/hooks/useEndlessScrollList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import { render, queryByAttribute, renderHook } from "@testing-library/react"
+import { useEndlessScrollList } from "./useEndlessScrollList"
+
+const intersectionObserverMock = () => ({
+  observe: () => null,
+  disconnect: () => null,
+})
+window.IntersectionObserver = vi.fn().mockImplementation(intersectionObserverMock)
+
+// TODO: add tests for the loading indicator which currently is not implemented do tue intersection observer can't be reproduced in tests
```
Comment (andypf) [inline packages/ui-components/src/hooks/useEndlessScrollList.test.tsx:15]: shouldn’t we remove this line?

File: packages/ui-components/tsconfig.hooks.build.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "./tsconfig.build.json", // base that will be ovewrited
```
Comment (andypf) [inline packages/ui-components/tsconfig.hooks.build.json:2]: Comments aren’t actually allowed in JSON, so I’m surprised the formatter and linter aren’t flagging this issue.

**Review by andypf — COMMENTED** [review]
Preview looks good! I see you also adjusted the Rollup config and updated all imports across the repo—great job! Overall, everything looks solid from my perspective; just a few minor notes.

File: packages/ui-components/tsconfig.hooks.build.json (hunk 2 above)
Comment (barsukov) [reply packages/ui-components/tsconfig.hooks.build.json:2]: Or, chatgpt leftover :) could you also test this scroll in app?

**Review by andypf — DISMISSED** [review]
lgtm

**Review by andypf — APPROVED** [review]
LGTM

**Review by ArtieReus — CHANGES_REQUESTED** [review]
i want to review

**Review by ArtieReus — CHANGES_REQUESTED** [review]

File: packages/ui-components/src/hooks/useEndlessScrollList.tsx [hunk 3]
```diff
@@ -49,39 +59,47 @@ const useEndlessScrollList = (items, options = {}) => {
...
-        const content = scrollListItems.map(f)
+      map: (elements: (value: unknown, index: number, array: unknown[]) => React.JSX.Element) => {
+        const content = scrollListItems?.map<React.JSX.Element>(elements)
         return (
           <>
             {content}
             {isAddingItems && options?.showLoading !== false && (
-              <>{options?.loadingObject ? options.loadingObject : <span>Loading...</span>}</>
+              <>
+                {options?.loadingObject ? options.loadingObject : <span id="endlessScrollListLoading">Loading...</span>}
```
[the remark this answers is not in the document]
Comment (barsukov) [reply packages/ui-components/src/hooks/useEndlessScrollList.tsx:86]: so means it is a problem of usage in doop? was it the same behaviour in doop before this tool?

[the remark this answers is not in the document]
Comment (ArtieReus) [reply packages/ui-components/src/hooks/useEndlessScrollList.tsx:86]: > I can see the Loading indicator, at least in Supernova App!

it is about the Doop app. Already synchronised with @barsukov

**Review by ArtieReus — APPROVED** [review]
lgtm

**Review by andypf — APPROVED** [review]

**Pull request merged**